test-utils/src/main/groovy/io/micronaut/kubernetes/test/KubernetesOperations.groovy

```diff
@@ -240,10 +255,13 @@ class KubernetesOperations implements Closeable {
         // in case of headless service or ExternalName service do now wait
         if (!(service.spec.externalName)) {
             log.debug("Polling for Endpoints get ready ${service}")
-            new PollingConditions().within(10){
+            new PollingConditions().within(10) {
                 assert getClient(namespace).endpoints().withName(name).get()
             }
         }
+
+        // give it some time to receive the event
+        sleep(1000)
```

Comment:
isn't there a better synchronization mechanism?

Reply:
good catch, left it here while testing.. sure there's a way :)